### PR TITLE
Wanchain address checksum

### PR DESCRIPTION
**Pull request by ArtemGoryunov**
## Description

Added the ability to support different checksum for ethereum-base blockchains.

## Testing instructions

Run tests

## Types of changes

- Added `TWEthereumChecksumType` enum and unit test for Wanchain address checksum.
- Setup new version release

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.

### CODE REVIEW

File: swift/Sources/Addresses/EthereumAddressExtension.swift [hunk 1]
```diff
@@ -8,6 +8,10 @@ import Foundation
 
 /// Ethereum address.
 extension EthereumAddress: Address, Hashable {
+    public convenience init?(string: String) {
+        self.init(string: string, type: .eip55)
```
Comment (vikmeup) [inline swift/Sources/Addresses/EthereumAddressExtension.swift]: how to create wanchain address checksum from string?

File: src/Coin.cpp [hunk 2]
```diff
@@ -55,10 +55,12 @@ std::string TW::loadAddress(TWCoinType coin, const Data& data) {
     case TWCoinTypeThunderToken:
     case TWCoinTypeTomoChain:
     case TWCoinTypeVeChain:
-    case TWCoinTypeWanChain:
     case TWCoinTypeXDai:
         return Ethereum::Address(data).string();
 
+    case TWCoinTypeWanChain:
+        return Ethereum::Address(data, TWEthereumChecksumTypeWanchain).string();
```
Comment (vikmeup) [inline src/Coin.cpp]: + you need to add validate address handler too. either all lowercase or checksummed address

File: swift/Sources/Addresses/EthereumAddressExtension.swift (hunk 1 above)
Comment (ArtemGoryunov) [reply swift/Sources/Addresses/EthereumAddressExtension.swift]: `EthereumAddress(string: string, type: .wanchain)`

File: src/Coin.cpp (hunk 2 above)
Comment (ArtemGoryunov) [reply src/Coin.cpp]: `Ethereum::isValid` checks only size `data.size() == size`

Comment (alejandro-isaza) [reply src/Coin.cpp]: We should have a `WanchainAddress` instead of adding a flag to EthereumAddress

File: src/Ethereum/Address.h [hunk 3]
```diff
@@ -16,6 +17,9 @@ namespace TW {
 namespace Ethereum {
 
 class Address {
+private:
+    TWEthereumChecksumType checksumType;
```
Comment (alejandro-isaza) [inline src/Ethereum/Address.h]: 👎

**Comment by hewigovens** [comment]
@alejandro-isaza could you please add wanchain checksum back? I remember this approach is how we did in trust core

**Pull request closed without merging**